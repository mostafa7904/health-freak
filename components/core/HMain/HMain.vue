<template>
  <main class="h-main">
    <div class="h-main__wrap">
      <slot />
    </div>
  </main>
</template>

<script lang="ts">
import { defineComponent } from 'vue'

export default defineComponent({
  name: 'HealthFreakMain'
})
</script>

<style lang="scss">
@import '~/assets/variables.scss';

.h-main {
  display: flex;
  flex: 1 0 auto;
  max-width: 100%;
  transition: 0.2s cubic-bezier(0.4, 0, 0.2, 1);

  &__wrap {
    flex: 1 1 auto;
    max-width: 100%;
    position: relative;
  }
}
</style>
