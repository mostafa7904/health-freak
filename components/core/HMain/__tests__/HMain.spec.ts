// Libraries
import Vue from "vue";

// Components
import HMain from "../HMain.vue";

// Utilities
import { createLocalVue, mount, Wrapper } from "@vue/test-utils";

describe("HMain.vue", () => {
  let mountFunction: (options?: object) => Wrapper<Vue>;

  let localVue: typeof Vue;

  beforeEach(() => {
    localVue = createLocalVue();

    mountFunction = (options = {}) => {
      return mount(HMain, {
        localVue,
        ...options,
      });
    };
  });

  it("should render component and match snapshot", () => {
    const html = mountFunction().html();
    expect(html).toMatchSnapshot();
  });
});
