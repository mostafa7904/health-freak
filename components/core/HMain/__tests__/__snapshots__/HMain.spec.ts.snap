// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HMain.vue should render component and match snapshot 1`] = `
"<main class=\\"h-main\\">
  <div class=\\"h-main__wrap\\"></div>
</main>"
`;
